forgerock-openbanking-uk-aspsp-common/src/main/java/com/forgerock/openbanking/common/services/store/data/DataEventsService.java

```diff
@@ -0,0 +1,170 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.common.services.store.data;
+
+import com.forgerock.openbanking.common.model.data.FRDataEvent;
+import com.forgerock.openbanking.common.model.openbanking.forgerock.event.FREventNotification;
+import com.forgerock.openbanking.exceptions.OBErrorResponseException;
+import com.forgerock.openbanking.model.error.OBRIErrorResponseCategory;
+import com.forgerock.openbanking.model.error.OBRIErrorType;
+import com.google.common.base.Preconditions;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.Collection;
+
+/**
+ * Logical entity Service to provide logic on the data sent to and from rs-store (store-server) and rs-ui (portal-server) apis
+ */
+@Service
+@Slf4j
+public class DataEventsService {
+
+    private final RestTemplate restTemplate;
+
+    @Value("${rs-store.base-url}")
+    private String rsStoreRoot;
+
+    @Autowired
+    public DataEventsService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    /**
+     * Export all events
+     * @return {@link FREventNotification} collection
+     */
+    public Collection<FREventNotification> exportEvents() {
+        log.debug("Export all data events");
+        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(rsStoreRoot + "/api/data/events/all");
+        URI uri = builder.build().encode().toUri();
+        return restTemplate.exchange(uri, HttpMethod.GET, null, getParameterizedTypeReference()).getBody();
+    }
+
+    /**
+     * Export the events by tppId (tpp username)
+     * @param frDataEvent the entity body {@link FRDataEvent}
+     * @return {@link FREventNotification} collection
+     * @throws {@link OBErrorResponseException}
+     */
+    public Collection<FREventNotification> exportEventsByTppId(FRDataEvent frDataEvent) throws OBErrorResponseException {
+        validateTppId(frDataEvent);
+        log.debug("Export all data events by user {}", frDataEvent.getTppId());
+        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(rsStoreRoot + "/api/data/events");
```

Comment:
Perhaps define a constant for `api/data/events`?

Reply:
True! and done!